### ISSUE TITLE
bug: Persistent Cookie Storage Issue in iOS App

### ISSUE THREAD

**Issue by Nasmar** [ionic-team/capacitor#6809]
### Capacitor Version

```
Latest Dependencies:       

  @capacitor/cli: 5.1.1    
  @capacitor/core: 5.1.1   
  @capacitor/android: 5.1.1
  @capacitor/ios: 5.1.1    

Installed Dependencies:    

  @capacitor/cli: 5.0.4    
  @capacitor/core: 5.0.4   
  @capacitor/ios: 5.0.4    
  @capacitor/android: 5.0.4
```

### Platform(s)

iOS

### Current Behavior

We cannot achieve persistent storage of HttpOnly cookies in our iOS app using CapacitorCookies and CapacitorHttp plugins.

Our app successfully retains cookies and functions as intended under normal usage. However, when the app is terminated and restarted, it fails to remember the cookies, resulting in users being prompted to log in again. 

Note that our Android app functions correctly, retaining the cookies as intended.

We are utilizing the native Capacitor Http module to make API calls.

### Expected Behavior

When shutting the iOS app down, we expect that the app can retain the http only cookie to use our cookie authentication

### Code Reproduction

I have made an Sample Project here:
https://happy-sea-00ef83503.3.azurestaticapps.net/
It simply calls a endpoint that sets the cookie "MyCookie" as a GUID.

On Android and PC - if you close the browser application it remembers the same Cookie if you push the button again.

On iOS it looses the cookie and you need to set a new one.

Link to sample project code, if needed:
https://github.com/Nasmar/sample_project

```
async function callEndpoint () {
  const apiUrl = "https://myapi.net"
  const url = apiUrl + '/api/HelloWorld'

  const response = await fetch(url, {
    headers: {
      'Accept': 'application/json',
      'Content-Type': 'application/json'
    },
    method: "GET",
    credentials: "include",
    mode: "cors"
  })

  const resultAsJson = await response.json()
  console.log('callEndpoint ', resultAsJson)
  result.value = resultAsJson
}

</script>

<template>
  <h1>Sample Project</h1>
  <div>
    <button @click="callEndpoint">Call endpoint</button>
    <h4>Result: {{ result }}</h4>
  </div>
</template>

<style scoped>
</style>

```

capacitor.config.ts:
```
  appId: 'com.XXXX.app',
  appName: 'XXXX',
  webDir: 'dist',
  plugins: {
    CapacitorHttp: {
      enabled: true // This provides native http service support via patching fetch and XMLHttpRequest
    },
    CapacitorCookies: {
      enabled: true
    }
  }
```

Info.plist:
```
	<key>WKAppBoundDomains</key>
	<array>
        <string>XXXX.XXXX.XX</string>
	<string>XXXX.XXXX.XX/api/</string>
	<string>https://XXXX.XXXX.XX/</string>
        <string>https://XXXX.XXXX.XX/api/</string>
	</array>
```

### Other Technical Details

`npm --version` output: 9.6.7

`node --version` output: 20.3.0

`pod --version` output (iOS issues only): 1.12.1

**Comment by Nasmar** [comment]
Link to sample project if needed:
https://github.com/Nasmar/sample_project

**Comment by ItsChaceD** [comment]
`HttpOnly` cookies should be consistent across all platforms as of 5.2.3 (#6787). 

I tested the sample app provided on a physical device with the latest version of Capacitor (5.2.3) and found the discrepancy that you are noticing. The `Set-Cookie` response that your server is returning does not include an `Expires` or `Max-Age` property. Therefore the cookie will be treated as a session cookie and will only last until the session closes. 

In a browser context, this typically means when the browser is closed (not just the tab). 
However in a native app context, this differs between Android and iOS. On iOS, this simply means when the app is terminated. On Android, cookies are persistent by default and "session" cookies (ie.. cookies without an expiration set) are only cleared when manually triggering `webkitCookieManager.removeSessionCookies(...);`. 

TLDR: I opened a PR so that Android session cookies match the behavior on web and iOS.

**Comment by Nasmar** [comment]
This explains why it only worked on Android. I have now set an Expires attribute on my cookie, and it appears to be functioning correctly on both iOS and Android.

Thank you for the response.
